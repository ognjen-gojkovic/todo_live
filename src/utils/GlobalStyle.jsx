import { createGlobalStyle, css } from "styled-components";

const ResetCSS = css`
  // variables
  :root {
    --background: #14004b;
    --border: #0f0038;
    --contentBg: #22017c;
    --font: #eaee02;
  }

  *,
  *::after,
  *::before {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: sans-serif;
  }
`;

const BaseStyle = createGlobalStyle`
    ${ResetCSS}
`;

export const GlobalStyles = () => <BaseStyle />;
