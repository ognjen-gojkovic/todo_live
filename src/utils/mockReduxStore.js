import { render as RTLRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import TodosReducer from "../redux/reducers/Reducer.Todos";

function render(
  component,
  {
    initState,
    store = configureStore({
      reducer: { todosReducer: TodosReducer },
      initState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return RTLRender(component, { wrapper: Wrapper, ...renderOptions });
}

// re-export all from testing package
export * from "@testing-library/react";
// override render method
export { render };
