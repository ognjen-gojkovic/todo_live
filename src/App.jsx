import React from "react";
import styled from "styled-components";
import Todos from "./components/todos/Todos";
import { GlobalStyles } from "./utils/GlobalStyle";

const App = () => {
  const canvasRef = React.useRef(null);

  React.useEffect(() => {
    let canvasBody = canvasRef.current;

    let w;
    let h;
    let particles = [];

    const opts = {
      particleColor: "#0f0038",
      lineColor: "#3b3b3b",
      particleAmount: 30,
      defaultSpeed: 0.2,
      variantSpeed: 0.1,
      defaultRadius: 50,
      variantRadius: 20,
      linkRadius: 200,
    };
    let drawArea = canvasBody.getContext("2d");
    let delay = 200,
      tid,
      rgb = opts.lineColor.match(/\d+/g);

    let resizeReset = function () {
      // added + 10px because of blur effect on right side of the screen, so with this 10px width of
      // canvas is bigger then screen, and additionally is taken care of with overflow-x: hidden on parent component
      w = canvasBody.width = window.innerWidth + 10;
      h = canvasBody.height = window.innerHeight;
    };

    window.addEventListener("resize", function () {
      deBouncer();
    });

    let deBouncer = function () {
      clearTimeout(tid);
      tid = setTimeout(function () {
        resizeReset();
      }, delay);
    };

    const Particle = function (xPos, yPos) {
      this.x = Math.random() * w;
      this.y = Math.random() * h;
      this.speed = opts.defaultSpeed + Math.random() * opts.variantSpeed;
      this.directionAngle = Math.floor(Math.random() * 360);
      this.color = opts.particleColor;
      this.radius = opts.defaultRadius + Math.random() * opts.variantRadius;
      this.vector = {
        x: Math.cos(this.directionAngle) * this.speed,
        y: Math.sin(this.directionAngle) * this.speed,
      };
      this.update = function () {
        this.border();
        this.x += this.vector.x;
        this.y += this.vector.y;
      };
      this.border = function () {
        if (this.x >= w || this.x <= 0) {
          this.vector.x *= -1;
        }
        if (this.y >= h || this.y <= 0) {
          this.vector.y *= -1;
        }
        if (this.x > w) this.x = w;
        if (this.y > h) this.y = h;
        if (this.x < 0) this.x = 0;
        if (this.y < 0) this.y = 0;
      };
      this.draw = function () {
        drawArea.beginPath();
        drawArea.arc(this.x, this.y, this.radius, 0, Math.PI * 2);
        drawArea.closePath();
        drawArea.fillStyle = this.color;
        drawArea.fill();
      };
    };

    function setup() {
      particles = [];
      resizeReset();
      for (let i = 0; i < opts.particleAmount; i++) {
        particles.push(new Particle());
      }
      window.requestAnimationFrame(loop);
    }

    function loop() {
      window.requestAnimationFrame(loop);
      drawArea.clearRect(0, 0, w, h);
      for (let i = 0; i < particles.length; i++) {
        particles[i].update();
        particles[i].draw();
      }
    }
    resizeReset();
    setup();
  }, []);

  return (
    <AppStyled>
      <GlobalStyles />
      <canvas ref={canvasRef} id="canvas"></canvas>
      <Todos />
    </AppStyled>
  );
};

const AppStyled = styled.div`
  height: 100vh;
  width: 100%;
  display: flex;
  justify-content: center;
  position: relative;
  canvas {
    background: linear-gradient(
      135deg,
      var(--border) 0% 20%,
      var(--background) 45%,
      var(--contentBg) 100%
    );
  }

  overflow-x: hidden;
`;

export default App;
