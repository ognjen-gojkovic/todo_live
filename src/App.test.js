import { render, screen } from "./utils/mockReduxStore";
import App from "./App";

test("renders learn react link", () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});
