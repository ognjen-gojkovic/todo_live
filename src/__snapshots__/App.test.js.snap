// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="sc-hKMtZM bxWXOJ"
>
  <canvas
    height="768"
    id="canvas"
    width="1034"
  />
  <div
    class="sc-dkzDqf jYIucv"
  >
    <div
      class="filter__wrapper"
    >
      <div
        class="sc-gsnTZi fxPONL"
      >
        <form>
          <input
            autocomplete="off"
            id="todo"
            name="todo"
            type="text"
            value=""
          />
          <label
            class="label"
            for="todo"
          >
            Enter Todo
          </label>
          <button
            disabled=""
            type="submit"
          >
            Add
          </button>
        </form>
      </div>
      <ul
        class="sc-bczRLJ bGHBls"
      />
    </div>
  </div>
</div>
`;
